File: employee-importance/employee-importance.cpp
/*
// Definition for Employee.
class Employee {
public:
    int id;
    int importance;
    vector<int> subordinates;
};
*/

class Solution {
public:
    int getImportance(vector<Employee*> employees, int id) {
        Employee* val;
        for(int i=0;i<employees.size();i++){
            Employee* temp = employees[i];
            if(temp->id ==id){
                val = employees[i];
                break;
            }
        }
        int sum = 0;
        queue<Employee*>q;
        q.push(val);
        while(!q.empty()){
            Employee* u = q.front();
            q.pop();
            sum+=u->importance;
            for(int i=0;i<u->subordinates.size();i++){
                int val = u->subordinates[i];
                for(int j=0;j<employees.size();j++){
                    Employee* temp = employees[j];
                    if(temp->id==val){
                        q.push(temp);
                        break;
                    }
                }
            }
        }
        return sum;
    }
};

File: employee-importance/employee-importance.cs
/*
// Definition for Employee.
class Employee {
    public int id;
    public int importance;
    public IList<int> subordinates;
}
*/

class Solution {
    public int GetImportance(IList<Employee> employees, int id) {
        Employee s = null;
        
        for(int i=0;i<employees.Count;i++)
        {
            Employee temp = employees[i];
            if(temp.id==id){
                s = employees[i];
                break;
            }
        }
        
        
        int sum = 0;
        
        Queue<Employee> q = new Queue<Employee>();
        q.Enqueue(s);
        
        while(q.Count!=0)
        {
            Employee u = q.Peek();
            q.Dequeue();
            sum+=u.importance;
            
            for(int i=0;i<u.subordinates.Count;i++)
            {
                int valu = u.subordinates[i];
                for(int j=0;j<employees.Count;j++)
                {
                    Employee obj =employees[j];
                    if(obj.id==valu)
                    {
                        q.Enqueue(obj);
                        break;
                    }
                }
            }
        }
        
        return sum;
    }
}